contracts/EpochTokenLocker.sol

```diff
@@ -59,28 +59,32 @@ contract EpochTokenLocker {
     }
 
     function requestWithdraw(address token, uint amount) public {
+        // first process old pendingWithdraw, as otherwise balances might increase for currentBatchId - 1
+        if (hasValidWithdrawRequest(msg.sender, token)) {
+            withdraw(token, msg.sender);
+        }
         balanceStates[msg.sender][token].pendingWithdraws = PendingFlux({ amount: amount, stateIndex: getCurrentBatchId() });
         emit WithdrawRequest(msg.sender, token, amount, getCurrentBatchId());
     }
 
-    function withdraw(address token) public {
-        updateDepositsBalance(msg.sender, token); // withdrawn amount might just be deposited before
+    function withdraw(address token, address owner) public {
```

Comment:
Why this change? Can't you keep using `msg.sender` here?

Reply:
If the solution submitter is doing the withdraw for another user with an open withdrawalRequest, then I must be enabled for anyone.